src/server.c

```diff
@@ -1043,6 +1043,8 @@ void clientsCron(void) {
     }
 }
 
+#define DATABASE_REHASH_TIME_PERC 1 /* Max % of CPU to use for databases rehash. */
```

Comment:
few things i wonder about:
1. why choosing 1%. as you explained, it's because it was previously set to 1ms, while the default hz is 10 (100 per seconds).
2. i suppose it could be interesting to increase this if the server is mostly idle (in which case the damage to traffic is less of a concern), maybe we can some day do that as an aftermath of #11963
3. active expiry is set by default for some 25% CPU, and can only be made more demanding.
4. and active defrag has a dynamic intensity based on fragmentation (default from 1% to 25%)

@madolson @soloestoy i wonder if you have anything to share about various concerns regarding this constant.

Reply:
Actually my code also references expire, but I don't think the two need to use the same mechanism.
25% of expire's cpu is the worst case, but rehash is **always** 1%. There are two scenarios when rehash occurs, expanding or shrinking. 
* For expanding, usually when the user pressure increases, the load itself is relatively heavy, the rehash in serverCron is actually dispensable. What we need to avoid at this time is the performance degradation caused by rehash, especially when executing commands such as mset,mget and msetnx.
* For shrinking, usually when the user pressure decreases, it is necessary to rely on the serverCron to speed up rehash and release memory space(but at this time the memory space is sufficient).
 it's may be useful for some scenes that increase the rehash when the server is mostly idle, but I think it might be over-engineered IMO

Reply:
i'm not certain your logic about heavy load when expanding and low load when shrinking is necessarily the typical case.
deletions are load just the same as insertions, and either way there could be high load of overwrites or reads regardless of shrinking or expanding.
maybe you're referring to expiration on a stale server?
@ofirluzon maybe you can provide your view on that?